app/layout/community/published-projects/detail/map/component.tsx

```diff
@@ -2,46 +2,57 @@ import React, {
   useCallback, useEffect, useMemo, useState,
 } from 'react';
 
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 
 import { useRouter } from 'next/router';
 
+import { setLayerSettings } from 'store/slices/projects/[id]';
+
 import PluginMapboxGl from '@vizzuality/layer-manager-plugin-mapboxgl';
 import { LayerManager, Layer } from '@vizzuality/layer-manager-react';
 import { AnimatePresence, motion } from 'framer-motion';
 import { useSession } from 'next-auth/client';
 
-import { usePUGridLayer } from 'hooks/map';
-import { usePublishedProject } from 'hooks/projects';
+import { useLegend, usePUGridLayer } from 'hooks/map';
+import { useProject } from 'hooks/projects';
 import { useScenarios } from 'hooks/scenarios';
 
+import HelpBeacon from 'layout/help/beacon';
+import { ScenarioSidebarTabs } from 'layout/scenarios/show/sidebar/types';
+
 import Map from 'components/map';
 import Controls from 'components/map/controls';
 import FitBoundsControl from 'components/map/controls/fit-bounds';
 import ZoomControl from 'components/map/controls/zoom';
+import Legend from 'components/map/legend';
+import LegendItem from 'components/map/legend/item';
+import LegendTypeBasic from 'components/map/legend/types/basic';
+import LegendTypeChoropleth from 'components/map/legend/types/choropleth';
+import LegendTypeGradient from 'components/map/legend/types/gradient';
+import LegendTypeMatrix from 'components/map/legend/types/matrix';
 
 export interface PublishedProjectMapProps {
 }
 
 export const PublishedProjectMap: React.FC<PublishedProjectMapProps> = () => {
-  const [viewport, setViewport] = useState({});
-  const [bounds, setBounds] = useState(null);
+  const [open, setOpen] = useState(false);
   const [session] = useSession();
```

Comment:
In the future we would need to work without session. At the moment i didn't manage to make Mapbox API works without sending auth .

Reply:
You are right... the endpoint to get the layers is authenticated. We will ask to remove that